### ISSUE TITLE
use of variables in clickRoutines

### ISSUE THREAD

**Issue by larkob** [ArnoldSmith86/virtualtabletop#305]
TLDR; I propose a new shorthand to `applyVariables` that aims to improve readability, in particular for consecutive computations.

When discussing PR#293 (operations in applyVariables), it became apparent that a) I didn't clearly explain the problem I wanted to solve, and b) the use of applyVariables is difficult already, so extending that functionality even further would potentially complicate rather than simplify things. So, I'm now looking for an alternative approach.

Let me start with two examples for the kind of problems I wanted to address:
1) determine a random value between 1-100, expressed computationally as sth. like `round(random() *100) +1`
2) determine the last letter of a string, expressed computationally as sth. like `str.charAt(str.length -1)`

My proposal is to introduce a shorthand `"{variablename}"` for applyVariables, which would look like this for example 1 (new approach to the right):
![image](https://user-images.githubusercontent.com/35028592/112125382-a949c600-8bc3-11eb-9d56-f0a1217f0871.png)

This shorthand becomes even more helpful when more than one variable needs to be applied, such as at the end of example 2 (new approach to the right):
![image](https://user-images.githubusercontent.com/35028592/112126456-c337d880-8bc4-11eb-93b1-13628fada738.png)

**Comment by rogerl50** [comment]
I love this. Code with `applyVariables` is almost impossible to read.

**Comment by ArnoldSmith86** [comment]
If you introduce ambiguity and string interpretation (`{...}`) anyway, why not go further?

Second example:
```JSON
"clickRoutine": [
  "myPosition = myString length",
  "myPosition = myPosition - 1",
  "myResult = myString charAt myPosition"
]
```

With `'foo bar'` for string literals.

**Comment by rogerl50** [comment]
I assume there is a security reason why you can't just allow a list of Javascript statements in the `clickRoutine` so that the example in the comment above would look more natural?

**Comment by ArnoldSmith86** [comment]
We didn't exactly research all the consequences but I'm pretty sure we all agree that one player shouldn't be allowed to execute arbitrary Javascript code on another player's machine.

**Comment by larkob** [comment]
Maybe I should have used a different symbol such as $variablename instead of the brackets to make this clear: With "{variablename}", I'm not suggesting to parse a string using RegEx. Variable values can be integer or null, so with `"operand1": "{varThatReturnsNull}"`, the value of operand1 should be null, not an empty string.

I'm not against expression parsing, but that involves a whole different level of coding.

**Comment by ArnoldSmith86** [comment]
I understand that. But that feature still means "there is no way to use string '{foo}'". And you would need to cut the string into parts. If you do that anyway, why is it "a whole different level of coding"?

**Comment by larkob** [comment]
I would simply be removing the brackets, that's it. And yes, your suggestion of a shorthand to set a variable looks like a great shorthand for COMPUTE, but you'd still need to apply the variable.
```
{
  "clickRoutine": [
    "myPosition = myString length",
    "myPosition = myPosition - 1",
    "myResult = myString charAt myPosition",
    {
      "func": "LABEL",
      "property": "text",
      "value": "{myResult}"
    }
  ]
}
```

**Comment by ArnoldSmith86** [comment]
Yes, we can do that as well. But I would also suggest that the variable names can also be widget properties. Something like `myLabelID->text = myString charAt myPosition`.

I think I'll just prepare a PR for that stuff so we can play around with it. Looks like fun.
